### PR TITLE
Dialog+search

### PR DESCRIPTION
**Pull request by HenryZou1**
Dialog+search+taskqueue fixed and intergrated

### CODE REVIEW

File: server/crawlers/twitter.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const getTwitterData = async (query) => {
       "tweet.fields": "entities,created_at,public_metrics",
       expansions: "author_id",
     });
-
+    if (data == undefined) return "none";
```
Comment (shahgrishma123) [inline server/crawlers/twitter.js:36]: use bool instead of string "none"

File: server/utils/taskQueue.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const bull = require("bull");
+const { getAllCompanyName } = require("../models/user-model");
+const searchQueue = new bull("searchQueue", {
+  redis: { port: 6379, host: "127.0.0.1" },
```
Comment (shahgrishma123) [inline server/utils/taskQueue.js:4]: It is better to move port and host variable value in the .env file.

File: server/utils/taskQueue.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+});
+const { redditSearch, getReddit } = require("../crawlers/reddit");
+const { getAndStoreTwitterData } = require("../crawlers/twitter");
+const User = require("../models/user-model");
+
+const companyQueue = new bull("companyQueue", {
```
Comment (shahgrishma123) [inline server/utils/taskQueue.js:10]: It is better to create a method that can create a bull queue and move new bull line into that function.

File: server/utils/taskQueue.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  User.getAllCompanyName().then((data) =>
+    data.forEach((element) => {
+      searchQueue.add({ company: element });
+    })
+  );
+}
+
+//, {repeat: {cron: '*/1 * * * *'}
+function start() {
+  companyQueue.add({}, { repeat: { cron: "*/1 * * * *" }, jobid: "companies" });
```
Comment (shahgrishma123) [inline server/utils/taskQueue.js:35]: It is better to have cron expression into the config file or .env file.

File: server/utils/taskQueue.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+});
+const { redditSearch, getReddit } = require("../crawlers/reddit");
+const { getAndStoreTwitterData } = require("../crawlers/twitter");
+const User = require("../models/user-model");
+
+const companyQueue = new bull("companyQueue", {
+  redis: { port: 6379, host: "127.0.0.1" },
+});
+searchQueue.process(function (job, done) {
+  console.log(job.data.company);
```
Comment (shahgrishma123) [inline server/utils/taskQueue.js:14]: Remove console.log from this.

**Review by shahgrishma123 — CHANGES_REQUESTED** [review]

File: client/src/components/Mention/Mention.js [hunk 6]
```diff
@@ -26,6 +28,17 @@ const useStyles = makeStyles((theme) => ({
     flexShrink: 0,
     flexGrow: 0,
   },
+  dialogPaper: {
+    minHeight: "80vh",
+    maxHeight: "80vh",
+    minWidth: "80vw",
+    maxWidth: "80vw",
```
Comment (johnsciutto) [inline client/src/components/Mention/Mention.js:35]: Did just doing `height: 80vh` and `width: 80vw` not work?

File: server/utils/taskQueue.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+const bull = require("bull");
+const { getAllCompanyName } = require("../models/user-model");
+const searchQueue = new bull("searchQueue", {
+  redis: { port: 6379, host: "127.0.0.1" },
+});
+const { redditSearch, getReddit } = require("../crawlers/reddit");
```
Comment (shahgrishma123) [inline server/utils/taskQueue.js:6]: I didn't see any use of getReddit variable.

**Review by shahgrishma123 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**